lib/new_relic/agent/instrumentation/async_http.rb

```diff
@@ -0,0 +1,31 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative 'async_http/instrumentation'
+require_relative 'async_http/chain'
+require_relative 'async_http/prepend'
+
+DependencyDetection.defer do
+  named :'async_http'
+
+  depends_on do
+    # The class that needs to be defined to prepend/chain onto. This can be used
+    # to determine whether the library is installed.
+    defined?(Async::HTTP)
+    # Add any additional requirements to verify whether this instrumentation
+    # should be installed
+  end
+
+  executes do
+    NewRelic::Agent.logger.info('Installing async_http instrumentation')
+
+    require 'async/http/internet'
+    require 'new_relic/agent/http_clients/async_http_wrappers'
+    if use_prepend?
+      prepend_instrument Async::HTTP::Internet, NewRelic::Agent::Instrumentation::AsyncHttp::Prepend
```

Comment:
Do we anticipate instrumenting any additional `Async::HTTP` classes in future? If not, this is great as-is. If so, maybe our own class name should include `Internet` in its name.

Reply:
I don't think so, I don't see any reason we'd need to intrument anything else since all the http requests will go through this instrumentation